### PR TITLE
[PORT] RCD ore silo link

### PR DESCRIPTION
**Pull request by Moccha-Bee**
Ports the RCD ore silo link from tg

## About The Pull Request
This port adds the RCD ore silo link upgrade from TG to Bee.
From:tgstation/tgstation/pull/45607

## Why It's Good For The Game

My back hurts from carrying so many sheets of reinforced glass...

## Changelog
:cl:
add: Finally the RCD can be directly linked to the Ore Silo.
/:cl:

### CODE REVIEW

File: code/game/objects/items/RCD.dm [hunk 1]
```diff
@@ -29,24 +29,32 @@ RLD
 	var/datum/effect_system/spark_spread/spark_system
 	var/matter = 0
 	var/max_matter = 100
-	var/sheetmultiplier	= 4 //Controls the amount of matter added for each glass/iron sheet, triple for plasteel
-	var/plasteelmultiplier = 3 //Plasteel is worth 3 times more than glass or iron
+	var/sheetmultiplier	= 4 //Controls the amount of matter added for each glass/metal sheet, triple for plasteel
```
Comment (Kerbin-Fiber) [inline code/game/objects/items/RCD.dm]: All the metal changes need to be changed to iron.

File: code/game/objects/items/RCD.dm [hunk 2]
```diff
@@ -29,24 +29,32 @@ RLD
...
+	var/rglassmultiplier = 1.5 //One metal sheet, half a glass sheet
 	var/no_ammo_message = "<span class='warning'>The \'Low Ammo\' light on the device blinks yellow.</span>"
 	var/has_ammobar = FALSE	//controls whether or not does update_icon apply ammo indicator overlays
 	var/ammo_sections = 10	//amount of divisions in the ammo indicator overlay/number of ammo indicator states
 	var/upgrade = FALSE
+	var/datum/component/remote_materials/silo_mats //remote connection to the silo
+	var/silo_link = FALSE //switch to use internal or remote storage
 
-/obj/item/construction/Initialize()
+/obj/item/construction/Initialize(mapload)
```
Comment (Kerbin-Fiber) [inline code/game/objects/items/RCD.dm]: take out `mapload`.

**Review by Kerbin-Fiber — CHANGES_REQUESTED** [review]
Seems like you took the files from /tg/ and slapped em in there and adjusted them to work, with some oversight.

File: code/game/objects/items/RCD.dm (hunk 1 above)
Comment (Kerbin-Fiber) [reply code/game/objects/items/RCD.dm]: in the comments

File: code/game/objects/items/RCD.dm (hunk 2 above)
Comment (Moccha-Bee) [reply code/game/objects/items/RCD.dm]: I changed it.

File: code/game/objects/items/RCD.dm (hunk 1 above)
Comment (Moccha-Bee) [reply code/game/objects/items/RCD.dm]: Replaced all metal with iron.

**Comment by MarkSuckerberg** [comment]
You've tested this, right?

**Comment by Moccha-Bee** [comment]
Yes but i noticed 2 problems. 1 the i accidently removed the option to build ladders and the advanced design rcd disk is useless i will remove it and fix the ladder problem.

**Comment by Moccha-Bee** [comment]
Okay fixed the ladder being missing aand the adavanced design disk can no longer be built or researched from the RD console. Its ready to be merged i guess.

File: code/game/objects/items/RCD.dm [hunk 3]
```diff
@@ -37,16 +37,24 @@ RLD
...
 
 /obj/item/construction/Initialize()
 	. = ..()
 	spark_system = new /datum/effect_system/spark_spread
 	spark_system.set_up(5, 0, src)
 	spark_system.attach(src)
+	if(upgrade & RCD_UPGRADE_SILO_LINK)
+		silo_mats = AddComponent(/datum/component/remote_materials, "RCD", FALSE)
 
 /obj/item/construction/examine(mob/user)
```
Comment (ike709) [inline code/game/objects/items/RCD.dm:51]: Have you tested this? TG changed how examine works and AFAIK we don't have those changes.

File: code/game/objects/items/RCD.dm [hunk 4]
```diff
@@ -315,7 +356,6 @@ RLD
 		"Medical" = get_airlock_image(/obj/machinery/door/airlock/medical),
 		"Research" = get_airlock_image(/obj/machinery/door/airlock/research),
 		"Freezer" = get_airlock_image(/obj/machinery/door/airlock/freezer),
-		"Science" = get_airlock_image(/obj/machinery/door/airlock/science),
```
Comment (ike709) [inline code/game/objects/items/RCD.dm]: ?

File: code/game/objects/items/RCD.dm [hunk 5]
```diff
@@ -370,8 +409,6 @@ RLD
 						airlock_type = /obj/machinery/door/airlock/research
 					if("Freezer")
 						airlock_type = /obj/machinery/door/airlock/freezer
-					if("Science")
```
Comment (ike709) [inline code/game/objects/items/RCD.dm]: ?

File: code/game/objects/items/RCD.dm (hunk 3 above)
Comment (Moccha-Bee) [reply code/game/objects/items/RCD.dm:51]: Yes and it was working fine last time i tested.

File: code/game/objects/items/RCD.dm (hunk 4 above)
Comment (Moccha-Bee) [reply code/game/objects/items/RCD.dm]: I din't cherrypick this file due to alot of errors but i will readd the deleted science airlock lines.

**Comment by ike709** [comment]
Undo the science airlock changes.

File: code/game/objects/items/RCD.dm [hunk 6]
```diff
@@ -334,7 +375,6 @@ RLD
 		"Command" = get_airlock_image(/obj/machinery/door/airlock/command/glass),
 		"Medical" = get_airlock_image(/obj/machinery/door/airlock/medical/glass),
 		"Research" = get_airlock_image(/obj/machinery/door/airlock/research/glass),
-		"Science" = get_airlock_image(/obj/machinery/door/airlock/science/glass),
```
Comment (ike709) [inline code/game/objects/items/RCD.dm]: 🤔

Comment (Moccha-Bee) [reply code/game/objects/items/RCD.dm]: Changed. Sorry for the mistakes

**Pull request merged**